pilot/pkg/serviceregistry/kube/controller/ambient/multicluster.go

```diff
@@ -658,3 +815,84 @@ func wrapObjectWithCluster[T any](clusterID cluster.ID) func(obj T) config.Objec
 		return config.ObjectWithCluster[T]{ClusterID: clusterID, Object: &obj}
 	}
 }
+
+func (a *index) createNetworkGatewayWorkload(networkGateway NetworkGateway, meshCfg *MeshConfig,
+) (*model.WorkloadInfo, error) {
+	address, err := netip.ParseAddr(networkGateway.Addr)
+	if err != nil {
+		return nil, err
+	}
+	wl := workloadapi.Workload{
+		Uid:            networkGateway.ResourceName(),
+		Name:           networkGateway.ResourceName(),
+		Namespace:      networkGateway.Source.Namespace,
+		Network:        networkGateway.Network.String(),
+		TunnelProtocol: workloadapi.TunnelProtocol_HBONE,
+		TrustDomain:    pickTrustDomain(meshCfg),
+		ServiceAccount: networkGateway.ServiceAccount.Name,
+		Capacity:       &wrapperspb.UInt32Value{Value: 1},
+		WorkloadType:   workloadapi.WorkloadType_DEPLOYMENT, // TODO(stevenjin8): What is the correct type here?
```

Comment:
I think we decided to always use Pod. @howardjohn can correct me, but I think it only matters for telemetry